hazelcast/src/main/java/com/hazelcast/spi/impl/SlowOperationLog.java

```diff
@@ -208,8 +211,10 @@ public JsonObject toJson() {
         public void fromJson(JsonObject json) {
             id = getInt(json, "id");
             startedAt = getLong(json, "startedAt");
-            durationNanos = getLong(json, "duration");
             lastAccessNanos = System.nanoTime();
+
+            // write durationMs last to enforce a write barrier
+            durationMs = getInt(json, "durationMs");
```

Comment:
I'll check the code tomorrow. But I'm talking about lastAccessNanos.

The fact that you are trying to play with fences is not a right move.